scipy/interpolate/tests/test_polyint.py

```diff
@@ -408,5 +417,106 @@ def test_all_zeros(self):
         assert_equal(pch(xx), 0.)
 
 
+class TestCubicSpline(object):
+    def test_general(self):
+        # This test can be reproduced in Octave/MATLAB with the following code:
+        # x = [-1, 0, 0.5, 2, 4, 4.5, 5.5, 9];
+        # y = [0, -0.5, 2, 3, 2.5, 1, 1, 0.5];
+        # x_eval = [-0.5, 0.1, 1, 3, 4.1, 5, 7];
+        # y_true = spline(x, y, x_eval);
+
+        x = np.array([-1, 0, 0.5, 2, 4, 4.5, 5.5, 9])
+        y = np.array([0, -0.5, 2, 3, 2.5, 1, 1, 0.5])
+
+        x_test = np.array([-0.5, 0.1, 1, 3, 4.1, 5, 7])
+        y_true = np.array([
+            -2.111801336869092, 0.019677035288960, 3.164826064645384,
+            3.384966086626085, 2.210195906837064, 0.464510284120531,
+            4.956649059572205])
+
+        cs = CubicSpline(x, y)
+        assert_allclose(cs(x_test), y_true, rtol=1e-12)
+
+        Y = np.vstack((y, y)).T
+        cs = CubicSpline(x, Y)
+        Y_true = np.vstack((y_true, y_true)).T
+        assert_allclose(cs(x_test), Y_true, rtol=1e-12)
+
+        Y = np.vstack((y, y))
+        cs = CubicSpline(x, Y, axis=1)
+        Y_true = np.vstack((y_true, y_true))
+        assert_allclose(cs(x_test), Y_true, rtol=1e-12)
+
+        Y = np.empty((2, 2, 8))
+        Y[0, 0, :] = y
+        Y[0, 1, :] = y
+        Y[1, 0, :] = y
+        Y[1, 1, :] = y
+        cs = CubicSpline(x, Y, axis=-1)
+        Y_true = np.empty((2, 2, 7))
+        Y_true[0, 0, :] = y_true
+        Y_true[0, 1, :] = y_true
+        Y_true[1, 0, :] = y_true
+        Y_true[1, 1, :] = y_true
+        assert_allclose(cs(x_test), Y_true, rtol=1e-12)
+
+    def test_corner_cases(self):
+        x = np.array([-1, 0, 1])
+        y = np.array([2, 0, 1])
+        x_test = np.array([-0.5, 0.5])
+        y_true = np.array([5/8, 1/8])
+        cs = CubicSpline(x, y)
+        assert_allclose(cs(x_test), y_true, rtol=1e-12)
+
+        x = np.array([-1, 1])
+        y = np.array([-1, 3])
+        y_true = np.array([0, 2])
+        cs = CubicSpline(x, y)
+        assert_allclose(cs(x_test), y_true, rtol=1e-12)
+
+    def _check_continuity(self, S, tol=1e-7):
+        c = S.c
+        dx = np.diff(S.x)[:-1]
+        assert_allclose(c[3, 1:], c[0, :-1] * dx**3 + c[1, :-1] * dx**2 +
+                        c[2, :-1] * dx + c[3, :-1], rtol=tol, atol=tol)
+        assert_allclose(c[2, 1:], 3 * c[0, :-1] * dx**2 + 2 * c[1, :-1] * dx +
+                        c[2, :-1], rtol=tol, atol=tol)
+        assert_allclose(c[1, 1:], 3 * c[0, :-1] * dx + c[1, :-1],
+                        rtol=tol, atol=tol)
```

Comment:
Minor implementation detail, of course. I'd expect something like this under the name of 'check_continuity', any reason to not do, e.g., 
```
for j in (0, 1, 2):
   assert_allclose(S(x[1:-1] - 1e-12, j), 
                   S(x[1:-1] + 1e-12, j), atol=1e-10)
```